packages/starlight/layout/Page.astro

```diff
@@ -45,6 +45,7 @@ const tocConfig = !hasSidebar
 	: config.tableOfContents;
 const hasToC = Boolean(tocConfig);
 const hasHero = Boolean(entry.data.hero);
+const pagefindEnabled = entry.slug !== '404' && entry.data.pagefind !== false;
```

Comment:
Revisiting this code reminded me we never accounted for translated 404s with this code. We can fix that by adding a test for slugs ending in `/404` like `es/404` etc.

```suggestion
const pagefindEnabled = entry.slug !== '404' && !entry.slug.endsWith('/404') && entry.data.pagefind !== false;
```